crates/sui-types/src/messages_checkpoint.rs

```diff
@@ -112,27 +112,19 @@ pub enum CheckpointRequestType {
 }
 
 #[derive(Clone, Debug, Serialize, Deserialize)]
-pub struct CheckpointResponse {
-    // The response to the request, according to the type
-    // and the information available.
-    pub info: AuthorityCheckpointInfo,
-    // If the detail flag in the request was set, then return
-    // the list of transactions as well.
-    // If the request is AuthenticatedCheckpoint, the detail is for the returned authenticated
-    // checkpoint; if the request is CheckpointProposal, the detail is for the returned proposal.
-    pub detail: Option<CheckpointContents>,
-}
-
-#[derive(Clone, Debug, Serialize, Deserialize)]
-pub enum AuthorityCheckpointInfo {
-    AuthenticatedCheckpoint(Option<AuthenticatedCheckpoint>),
+pub enum CheckpointResponse {
+    AuthenticatedCheckpoint {
+        checkpoint: Option<AuthenticatedCheckpoint>,
```

Comment:
nit: One of the way to improve this type system can be adding `NotFound` to CheckpointResponse and then reviewing all those `Options`. For example AuthenticatedCheckpoint::checkpoint would not need to be an Option.

Also probably CheckpointProposal::proposal and CheckpointProposal::prev_cert too would not be optional then.

Maybe its a personal preference but this would help to avoid unreachable permutations of Options, like CheckpointProposal::prev_cert and CheckpointProposal::proposal are either both None or both Some

Reply:
Good suggestion. That would indeed make it clearer.
We can probably define NotFound as a newtype wrapper of Option and deref to Option

Reply:
Just to clarify - is this something you plan to do in this PR?(e.g. do we need to wait for those changes before reviewing?)

Reply:
No that would not be in this PR